data/topic-filter.go

```diff
@@ -0,0 +1,222 @@
+package data
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	errs "github.com/ONSdigital/dp-frontend-search-controller/apperrors"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// Filter represents information of filters selected by user
+type TopicFilter struct {
+	Query           []string `json:"query,omitempty"`
+	LocaliseKeyName []string `json:"localise_key,omitempty"`
+}
+
+// TopicCategory represents all the search categories in search page
+type TopicCategory struct {
+	LocaliseKeyName string             `json:"localise_key"`
+	Count           int                `json:"count"`
+	ContentTypes    []TopicContentType `json:"content_types"`
```

Comment:
This surely should be `Topics` instead of `ContentTypes` and should be of type `[]Topic` and json tag should be `topics`?

```suggestion
	Topics    []Topic `json:"topics"`
```